### ISSUE TITLE
Stepping in debugger is too smooth

### ISSUE THREAD

**Issue by jrieken** [Microsoft/vscode#1453]
- have a breakpoint in a loop
- run and once hitting the breakpoint keep pressing play
- > notice that the feedback of resume and pause is not strong enough. It seems we animate/delay something which hides the state transitions.

**Comment by bpasero** [comment]
@isidorn feel free to change the label, not sure this is intentional behavior or not.

**Comment by isidorn** [comment]
The reason for this is that we change the state only when the response comes back from the server, not before. The state change triggers the icon change.

Another issue here is that sometimes the adapter sends the "stopped" event earlier then the respone for the 'continue' request. Due to this we only emit "continued" if we did not miss a stopped event. We do not emit straight away to reduce viewlet flickering. More details 
https://github.com/Microsoft/vscode/blob/release/0.10.6/src/vs/workbench/parts/debug/node/rawDebugSession.ts#L97

Bottom line this is currently a limitation which I do not plan to fix.
Feel free to reopen if you have some suggestions on how to improve this while still working around the issues explaind above 😇

**Comment by jrieken** [comment]
I would have not filed this otherwise. Just update the UI when the user interacts with it not when the server eventually returns a result.

**Comment by weinand** [comment]
@isidorn I just tried this on macOS and most of the time I see no "flickering" of the play arrow when keeping the F5 key pressed. So the "Stopped" button has not enough time to become visible before another "Continue" is issued.

Could we try to introduce a small delay receiving a "stopped" event?

**Comment by weinand** [comment]
fix it or close it.
